tempesta_fw/http.c

```diff
@@ -272,6 +278,12 @@ tfw_http_send_resp(TfwHttpReq *req, TfwStr *msg, const TfwStr *date)
 	tfw_http_resp_fwd(req, (TfwHttpResp *)hmresp);
 
 	return 0;
+mem_err:
+	TFW_DBG2("%s: Memory allocation error: conn=[%p]\n",
+		 __func__, req->conn);
+	ss_close_sync(req->conn->sk, true);
+	tfw_http_conn_msg_free((TfwHttpMsg *)req);
+	return -ENOMEM;
```

Comment:
Since this is another error, related to the inability to send a response to the client with information about the initial error - then yes, we should. Corrected.